pkg/api/vlabs/validate.go

```diff
@@ -354,6 +354,59 @@ func (a *Properties) validateMasterProfile() error {
 			return err
 		}
 	}
+
+	if m.IsVirtualMachineScaleSets() && a.OrchestratorProfile.OrchestratorType == Kubernetes {
+		version := common.RationalizeReleaseAndVersion(
+			a.OrchestratorProfile.OrchestratorType,
+			a.OrchestratorProfile.OrchestratorRelease,
+			a.OrchestratorProfile.OrchestratorVersion,
+			false,
+			false)
+		if version == "" {
+			return errors.Errorf("the following OrchestratorProfile configuration is not supported: OrchestratorType: %s, OrchestratorRelease: %s, OrchestratorVersion: %s. Please check supported Release or Version for this build of acs-engine", a.OrchestratorProfile.OrchestratorType, a.OrchestratorProfile.OrchestratorRelease, a.OrchestratorProfile.OrchestratorVersion)
+		}
+
+		sv, err := semver.Make(version)
+		if err != nil {
+			return errors.Errorf("could not validate version %s", version)
+		}
+		minVersion, err := semver.Make("1.10.0")
+		if err != nil {
+			return errors.New("could not validate version")
+		}
+		if sv.LT(minVersion) {
+			return errors.Errorf("VirtualMachineScaleSets are only available in Kubernetes version %s or greater. Please set \"orchestratorVersion\" to %s or above", minVersion.String(), minVersion.String())
+		}
+		// validation for instanceMetadata using VMSS with Kubernetes
+		minVersion, err = semver.Make("1.10.2")
+		if err != nil {
+			return errors.New("could not validate version")
+		}
+		if a.OrchestratorProfile.KubernetesConfig != nil && a.OrchestratorProfile.KubernetesConfig.UseInstanceMetadata != nil {
+			if *a.OrchestratorProfile.KubernetesConfig.UseInstanceMetadata && sv.LT(minVersion) {
+				return errors.Errorf("VirtualMachineScaleSets with instance metadata is supported for Kubernetes version %s or greater. Please set \"useInstanceMetadata\": false in \"kubernetesConfig\" or set \"orchestratorVersion\" to %s or above", minVersion.String(), minVersion.String())
+			}
+		}
+		if m.AvailabilityProfile == VirtualMachineScaleSets && m.StorageProfile == StorageAccount {
```

Comment:
Isn't this part of the condition redundant? We are already checking the same condition in the outer block if I am not mistaken.